lib/Sema/TypeCheckPropertyWrapper.cpp

```diff
@@ -580,7 +580,6 @@ PropertyWrapperBackingPropertyTypeRequest::evaluate(
 
   ASTContext &ctx = var->getASTContext();
   Type type = ctx.getSideCachedPropertyWrapperBackingPropertyType(var);
-  assert(type || ctx.Diags.hadAnyError());
```

Comment:
Instead of removing this assertion, could you insert
```
    if (binding->isInvalid())
      return Type();
```
at the end of `if (binding && binding->isInitialized(index))` branch?

Reply:
Ah, that looks like a cleaner solution. Applied it.